test/e2e/multihoming.go

```diff
@@ -792,6 +792,95 @@ var _ = Describe("Multi Homing", func() {
 						Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(MatchError(ContainSubstring("Connection timeout")))
 					})
 				})
+
+				When("a policy is provisioned", func() {
+					var clientPodConfig podConfiguration
+					allPodsSelector := map[string]string{}
+
+					const (
+						denyAllIngress = "deny-all-ingress"
+						allowAllEgress = "allow-all-egress"
+						denyAllEgress  = "deny-all-egress"
+					)
+
+					BeforeEach(func() {
+						clientPodConfig = podConfiguration{
+							name:        clientPodName,
+							namespace:   f.Namespace.Name,
+							attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+							labels:      allPodsSelector,
+						}
+						kickstartPod(cs, clientPodConfig)
+					})
+
+					DescribeTable("can communicate over a localnet secondary network from pod to gw", func(mnp *mnpapi.MultiNetworkPolicy) {
+						By("provisioning the multi-network policy")
+						_, err := mnpClient.MultiNetworkPolicies(clientPodConfig.namespace).Create(
+							context.Background(),
+							mnp,
+							metav1.CreateOptions{},
+						)
+						Expect(err).NotTo(HaveOccurred())
+
+						if mnp.Name != denyAllEgress {
+							Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(Succeed())
+						} else {
+							Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(Not(Succeed()))
+						}
+
+						By("deleting the multi-network policy")
+						Expect(mnpClient.MultiNetworkPolicies(clientPodConfig.namespace).Delete(
+							context.Background(),
+							mnp.Name,
+							metav1.DeleteOptions{},
+						)).To(Succeed())
+
+						By("asserting the *client* pod can contact the underlay service after deleting the policy")
+						Expect(connectToServer(clientPodConfig, underlayServiceIP, servicePort)).To(Succeed())
+					},
+						XEntry(
+							"minimal ingress denyall, ingress policy should have no impact on egress",
+							multiNetIngressDenyAllPolicy(
+								denyAllIngress,
+								secondaryNetworkName,
+								metav1.LabelSelector{
+									MatchLabels: allPodsSelector,
+								},
+							),
+							Label("BUG", "OCPBUGS-25928"),
+						),
+						Entry(
+							"ingress denyall, egress allow all, ingress policy should have no impact on egress",
+							multiNetIngressDenyAllEgressAllowAllPolicy(
+								denyAllIngress,
+								secondaryNetworkName,
+								metav1.LabelSelector{
+									MatchLabels: allPodsSelector,
+								},
+							),
+						),
+						Entry(
+							"egress allow all",
+							multiNetEgressAllowAllPolicy(
+								allowAllEgress,
+								secondaryNetworkName,
+								metav1.LabelSelector{
+									MatchLabels: allPodsSelector,
+								},
+							),
+						),
+						Entry(
+							"egress deny all",
```

Comment:
how about the reverse scenario, does egress deny all impact ingress?

Reply:
will be added indeed in this https://github.com/ovn-org/ovn-kubernetes/pull/4153
which check ingress traffic
this PR checks egress traffic (where the policies are either ingress deny all / egress deny or allow all)

Reply:
btw, same bug exists in the opposite direction, soon will update mentioned PR with the test

Reply:
btw, (almost) same bug exists in the reverse scenario, soon will update mentioned PR with the test
the weird thing is that the wa (adding ingress wildcard rule) doesnt work on the reverse scenario

Reply:
done on
https://github.com/ovn-org/ovn-kubernetes/pull/4153#issuecomment-1954583375